src/neural_mesh_simplification/losses/edge_crossing_loss.py

```diff
@@ -48,32 +59,15 @@ def find_nearest_triangles(
         k = min(
             self.k, centroids.shape[0]
         )  # Ensure k is not larger than the number of centroids
-        _, indices = knn(centroids, centroids, k=k)
-
-        # Reshape indices to [num_faces, k]
-        indices = indices.view(centroids.shape[0], k)
-
-        # Remove self-connections (triangles cannot be their own neighbor)
-        nearest = []
-        for i in range(indices.shape[0]):
-            neighbors = indices[i][indices[i] != i]
-            if len(neighbors) == 0:
-                nearest.append(torch.empty(0, dtype=torch.long))
-            else:
-                nearest.append(neighbors[: self.k - 1])
-
-        # Return tensor with consistent shape
-        if len(nearest) > 0 and all(len(n) == 0 for n in nearest):
-            nearest = torch.empty((len(nearest), 0), dtype=torch.long)
-        else:
-            nearest = torch.stack(nearest)
-        return nearest
+        g_knn = dgl.knn_graph(centroids, k)
```

Comment:
```suggestion
        g_knn = dgl.knn_graph(centroids, k, exclude_self=True)
```

If we want to avoid the triangle itself ranking as the closest, `exclude_self` should be set. We explicitly removed self-connections prior to this change.